tremor-script/src/grammar.lalrpop

```diff
@@ -23,7 +24,172 @@ use beef::Cow;
+        let mut fields = fields;
+        fields.reverse();
+        fields
+    },
+}
+
+DeployLinks_: DeployLinksRaw<'input> = {
+    <Sep<DeployLinks_, DeployLink, ",">> => <>
+}
+
+DeployLink: DeployLinkRaw<'input> = {
+    // FIXME TODO TBD
+    <start:@L> "connect" <from:DeployEndpoint> "to" <to:DeployEndpoint> <end:@L> => DeployLinkRaw { from, to }
+    // <start:@L> <name:Ident> => ArgumentDeclRaw::Required(name),
+    // <start:@L> <name:Ident> "=" <default_value:ExprImut> => ArgumentDeclRaw::Optional(name, default_value)
+}
+
+DeployEndpoint: DeployEndpointRaw<'input> = {
+    StringLiteral => DeployEndpointRaw::Legacy(<>),
+    ModularTarget => DeployEndpointRaw::Troy(<>),
+}
+
+/// List of expressions at the root of the script
+DeployModuleStmts: DeployStmtsRaw<'input> = {
+     <stmt:DeployModuleInnerStmt> ";" <list:DeployModuleStmts> => {
+        let mut list = list;
+        list.insert(0,stmt);
+        list
+     },
+     <item:DeployModuleInnerStmt> ";"? => vec![item],
+ }
+
+
+// In case we need to limit the content of deploy stmt
+DeployModuleInnerStmt: DeployStmtRaw<'input> = {
+    <m:DeployModuleStmt> => DeployStmtRaw::ModuleStmt(m),
+    // Pipelines - these define parameterized queries for deployment
+    <start:@L> "define" "pipeline" <id:Ident> <args:ArgSpecificationClause> <query:EmbeddedQuery> <end:@L> => DeployStmtRaw::PipelineDecl(PipelineDeclRaw { start, end, id: id.to_string(), query, args: args.0, params: args.1 }),
+    <start:@L> "define" "pipeline" <id:Ident> <query:EmbeddedQuery> <end:@L> => DeployStmtRaw::PipelineDecl(PipelineDeclRaw { start, end, id: id.to_string(), query, args: vec![], params: None }),
+    // Connectors - these define connectors for deployment
+    <start:@L> "define" <kind:ConnectorKind> "connector" <id:Ident> <args:ArgSpecificationClause> "end" <end:@L> => DeployStmtRaw::ConnectorDecl(ConnectorDeclRaw { start, end, id: id.to_string(), kind, args: args.0, params: args.1 }),
+    <start:@L> "define" <kind:ConnectorKind> "connector" <id:Ident> "end" <end:@L> => DeployStmtRaw::ConnectorDecl(ConnectorDeclRaw { start, end, id: id.to_string(), kind, args: vec![], params: None }),
+    // Flow - these define how pipelines and connectors are interconnected to form a unit deployment graph
+    <start:@L> "define" "flow" <id:Ident> <args:ArgSpecificationClause> <links:WithDeployLinks> "end" <end:@L> => DeployStmtRaw::FlowDecl(FlowDeclRaw { start, end, id: id.to_string(), args: args.0, params: args.1, links }),
+    <start:@L> "define" "flow" <id:Ident> <links:WithDeployLinks> "end" <end:@L> => DeployStmtRaw::FlowDecl(FlowDeclRaw { start, end, id: id.to_string(), args: vec![], params: None, links }),
+
+    // Allow imports of tremor-script definitions
+    Const => DeployStmtRaw::Expr(Box::new(<>)),
+    FnDecl => DeployStmtRaw::Expr(Box::new(ExprRaw::FnDecl(<>))),
+    Intrinsic => DeployStmtRaw::Expr(Box::new(ExprRaw::FnDecl(<>))),
+}
+
+
+EmbeddedQuery: QueryRaw<'input> = {
```

Comment:
this should probably use the modular subquery logic from Yatin once his PR is done

Reply:
That's the plan - i'll add a TODO so we have this captured

Reply:
TODO added